instrumentation/grpc-1.5/src/main/java/io/opentelemetry/auto/instrumentation/grpc/client/GrpcClientDecorator.java

```diff
@@ -28,12 +29,18 @@
       OpenTelemetry.getTracerProvider().get("io.opentelemetry.auto.grpc-1.5");
 
   public Span onClose(final Span span, final io.grpc.Status status) {
-
-    span.setAttribute("status.code", status.getCode().name());
-    span.setAttribute("status.description", status.getDescription());
-
-    onError(span, status.getCause());
+    super.onError(span, status.getCause());
     span.setStatus(GrpcHelper.statusFromGrpcStatus(status));
```

Comment:
If there is throwable, you will end up calling setStatus twice. And it seems like second one will override one set in `onError`

Reply:
It's a bit unclear - this is calling `super.onError` now. I added a comment but admit it's a bit unclear, couldn't come up with a way to refactor that would be clearer though. Any suggestion?

Reply:
maybe adding an `onError()` overload in `BaseDecorator` that also takes (otel) `Status` might help?

Reply:
Thanks for the idea, looks a bit better.